packages/runtime/container-runtime/src/opLifecycle/opGroupingManager.ts

```diff
@@ -26,16 +28,42 @@ function isGroupContents(opContents: any): opContents is IGroupedBatchMessageCon
 	return opContents?.type === OpGroupingManager.groupedBatchOp;
 }
 
+export interface OpGroupingManagerConfig {
+	readonly groupedBatchingEnabled: boolean;
+	readonly opCountThreshold: number;
+	readonly reentrantBatchGroupingEnabled: boolean;
+}
+
 export class OpGroupingManager {
 	static readonly groupedBatchOp = "groupedBatch";
+	private readonly logger;
 
-	constructor(private readonly groupedBatchingEnabled: boolean) {}
+	constructor(
+		private readonly config: OpGroupingManagerConfig,
+		logger: ITelemetryBaseLogger,
+	) {
+		this.logger = createChildLogger({ logger, namespace: "OpGroupingManager" });
+	}
 
 	public groupBatch(batch: IBatch): IBatch {
-		if (batch.content.length < 2 || !this.groupedBatchingEnabled) {
+		if (!this.shouldGroup(batch)) {
 			return batch;
 		}
 
+		assert(
+			batch.content.length >= this.config.opCountThreshold,
+			"Expecting to always group batches with op counts larger than the threshold",
```

Comment:
I'm not sure I follow the spirit of this assert. Is it checking that this.shouldGroup() logic works correctly? What if we decide to group only if it's above threshold and there is no reentrancy (i.e. replace OR with AND)?
I'd probably not have that assert. :)